oddt/shape.py

```diff
@@ -0,0 +1,250 @@
+    donor_shape = common_usr(molecule, ctd, cst, fct, ftf, 'isdonor')[0]
+
+    cat_shape = np.hstack((all_atoms_shape, hydrophobic_shape,
+                           aromatic_shape, acceptor_shape, donor_shape))
+
+    return np.nan_to_num(cat_shape)
+
+
+def electroshape(mol):
+    """Computes shape descriptor based on
+    Armstrong, M. S. et al. ElectroShape: fast molecular similarity
+    calculations incorporating shape, chirality and electrostatics.
+    J Comput Aided Mol Des 24, 789-801 (2010).
+    http://dx.doi.org/doi:10.1007/s10822-010-9374-0
+
+    Aside from spatial coordinates, atoms' charges are also used
+    as the fourth dimension to describe shape of the molecule.
+
+        Parameters
+        ----------
+            mol : oddt.toolkit.Molecule
+
+        Returns
+        -------
+            shape_descriptor : numpy array, shape = (15)
+                               Array describing shape of molecule
+    """
+    if (mol.atom_dict["coords"] == 0).all():
+        raise Exception("Molecule needs 3D coordinates")
+
+    mi = 25  # scaling factor converting electron charges to Angstroms
+
+    four_dimensions = np.column_stack((mol.atom_dict["coords"], 
+                                       mol.atom_dict['charge'] * mi))
+
+    c1 = four_dimensions.mean(0)  # geometric centre of the molecule
+    distances_c1 = norm(four_dimensions - c1, axis=1)
+
+    c2 = four_dimensions[distances_c1.argmax()]  # atom position furthest from c1
+    distances_c2 = norm(four_dimensions - c2, axis=1)
+
+    c3 = four_dimensions[distances_c2.argmax()]  # atom position furthest from c2
+    distances_c3 = norm(four_dimensions - c3, axis=1)
+
+    vector_a = c2 - c1
+    vector_b = c3 - c1
+    vector_as = vector_a[:3]    # spatial parts of these vectors —
+    vector_bs = vector_b[:3]    # the first three coordinates
+    vector_c = (norm(vector_a) /
+                (2 * norm(np.cross(vector_as, vector_bs))))\
+        * np.cross(vector_as, vector_bs)
+
+    vector_c1s = c1[:3]
+
+    max_charge = np.array(np.amax(mol.atom_dict['charge']) * mi)
+    min_charge = np.array(np.amin(mol.atom_dict['charge']) * mi)
+
+    c4 = np.append(vector_c1s + vector_c, max_charge)
+    c5 = np.append(vector_c1s + vector_c, min_charge)
+
+    distances_c4 = norm(four_dimensions - c4, axis=1)
+    distances_c5 = norm(four_dimensions - c5, axis=1)
+
+    distances_list = [distances_c1, distances_c2, distances_c3,
+                      distances_c4, distances_c5]
+
+    shape_descriptor = np.zeros(15)
+
+    i = 0
+    for distances in distances_list:
+        mean = np.mean(distances)
+        shape_descriptor[0 + i] = mean
+        shape_descriptor[1 + i] = np.std(distances)
+        shape_descriptor[2 + i] = np.cbrt(np.sum(
+            ((distances - mean) ** 3) / distances.size))
+        i += 3
+
+    return shape_descriptor
+
+
+def similarity(mol1_shape, mol2_shape, ow=1., hw=1., rw=1., aw=1., dw=1.):
```

Comment:
Please rename to `usr_similarity`.
Either refer to all publications or to none - point to them in each USR's docs (in docs).